hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/TinyLfuBlockCache.java

```diff
@@ -177,6 +177,7 @@ public Cacheable getBlock(BlockCacheKey cacheKey,
       }
     } else if (updateCacheMetrics) {
       stats.hit(caching, cacheKey.isPrimary(), cacheKey.getBlockType());
+      value.retain();
```

Comment:
We should follow the same as in LRU cache no?
LruCachedBlock cb = map.computeIfPresent(cacheKey, (key, val) -> {
      // It will be referenced by RPC path, so increase here. NOTICE: Must do the retain inside
      // this block. because if retain outside the map#computeIfPresent, the evictBlock may remove
      // the block and release, then we're retaining a block with refCnt=0 which is disallowed.
      // see HBASE-22422.
      val.getBuffer().retain();
      return val;
    });
And other places